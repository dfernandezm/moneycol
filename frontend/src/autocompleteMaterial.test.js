import React from 'react';
import ReactDOM from 'react-dom';
import AutocompleteMaterial from './autocompleteMaterial';
import renderer from 'react-test-renderer';

it('renders Autocomplete without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<AutocompleteMaterial />, div);
  ReactDOM.unmountComponentAtNode(div);
});


it('renders Autocomplete with Country label', () => {
 
  const component = renderer.create(
    <AutocompleteMaterial/>,
  );

  let tree = component.toJSON();
  //To update the snapshot, run type u if you are still in watch mode, 
  //or simply delete the snapshot file under the __snapshots__ folder
  expect(tree).toMatchSnapshot();  

});

it('it initiates only 1 autocomplete materialize css', () => {
 
  const component = renderer.create(
    <AutocompleteMaterial/>
  );
  //TODO: internal behaviour, consider design of this
  expect(component.getInstance().autocompletes).toBeDefined();
});

xit('it renders value "Afganistan" when typing "afg"', () => {
 
  // const component = renderer.create(
  //   <AutocompleteMaterial/>
  // );
  
});