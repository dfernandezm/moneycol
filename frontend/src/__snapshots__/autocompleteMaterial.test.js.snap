// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Autocomplete with Country label 1`] = `
<div
  className="row"
>
  <div
    className="col s12"
  >
    <div
      className="row"
    >
      <div
        className="input-field col s12"
      >
        <i
          className="material-icons prefix"
        >
          textsms
        </i>
        <input
          className="autocomplete"
          id="autocomplete-input"
          type="text"
        />
        <label
          htmlFor="autocomplete-input"
        >
          Country
        </label>
      </div>
    </div>
  </div>
</div>
`;
